api/webhook/taskcluster.go

```diff
@@ -252,6 +251,31 @@ func verifySignature(message []byte, signature string, secret string) bool {
 	return hmac.Equal(messageMAC, expectedMAC)
 }
 
+func createAllRuns(log shared.Logger, transport http.RoundTripper, api, username, password string, urlsByBrowser map[string][]string) error {
+	client := &http.Client{Transport: transport}
+
+	errors := make(chan error, len(urlsByBrowser))
+	var wg sync.WaitGroup
+	wg.Add(len(urlsByBrowser))
+	for browser, urls := range urlsByBrowser {
+		go func(browser string, urls []string) {
+			defer wg.Done()
+			log.Infof("Reports for %s: %v", browser, urls)
+			err := createRun(client, api, username, password, urls)
+			if err != nil {
+				log.Errorf("%v", err)
+				errors <- err
+			}
+		}(browser, urls)
+	}
+	wg.Wait()
+	close(errors)
+	if _, ok := <-errors; ok {
+		return fmt.Errorf("error(s) occured when talking to %s", api)
```

Comment:
As we discussed, one possibility here is to concat strings that constitute the errors. I would say that's worth doing if:

1. This error string may get propagated to somewhere other than `stderr` (e.g., in a HTTP response body), and
1. Someone without access to our logs may find themselves reading (somewhere other than `stderr`) and wondering which runs succeeded and which runs failed.

Reply:
... in which case we should probably use `for ... range errors { ... }` to capture the error, and also make sure that the error strings contain information about which run/browser caused the error(s).

Reply:
Done.